### DISCUSSION TITLE
Problem after logging out

### DISCUSSION THREAD

**Discussion by flessaaco** [SteveG/underthebar#4] [General]
Hi, 

This app is a great idea, great job! Coming here after your post on Reddit.
I've managed to open it on my main account, then logged out to try on a new/fresh account (the new account has a dummy session recorded). After logging out, when I try to reopen the error below

Traceback (most recent call last):
  File "e:\Dropbox (Penn)\Documentos\4. Saúde\4. Workout e dieta\underthebar-main\underthebar.py", line 303, in <module>
    window = UnderTheBar(sillytheme)
  File "e:\Dropbox (Penn)\Documentos\4. Saúde\4. Workout e dieta\underthebar-main\underthebar.py", line 118, in __init__
    self.stacklayout.addWidget(Setting("orange"))
  File "e:\Dropbox (Penn)\Documentos\4. Saúde\4. Workout e dieta\underthebar-main\utb_page_setting.py", line 106, in __init__
    self.remoteworkoutsLabel = QLabel("Remote Workouts: "+str(workoutcount_data["data"]["workout_count"]))
TypeError: 'NoneType' object is not subscriptable

**Comment by flessaaco** [comment]
Found the issue. When I logged in with the new account, I had no pre-recorded workout session. That made the program crash. Upon trying to reopen the program, I already had a user temp file for this account, but no workout JSON. That was the root of the latter error.

Maybe to avoid this happening to people as they test around, it would be good to include an exception handling such that, if the program fails to load any JSON, it deletes all temp files for the current user and runs the program without any user signed in.
Additionally, as you get more regular users to your program, would be good to put an error handling to not having any profile picture.

Once again, amazing job with this program!

**Reply by SteveG** [reply]
Hey thanks for the feedback, great to hear you've had some success. Hope it's useful to you.

Yeah I definitely acknowledge it's not very polished. Programming is not my day job and I've only had some pretty mild interest from other people so it doesn't seem super worth investing too much effort into. I might have a look at your particular feedback though.

Feel free to let me know if you have a great idea for a new graph etc. I think the routine editor I added was entirely inspired by your reddit post.
